src/Microsoft.Data.Sqlite.Core/SqliteCommand.cs

```diff
@@ -510,21 +518,19 @@ private IEnumerable<sqlite3_stmt> PrepareAndEnumerateStatements()
             while (!string.IsNullOrEmpty(tail));
         }
 
-        private IEnumerable<sqlite3_stmt> ResetAndEnumerateStatements()
+        private void DisposePreparedStatements()
         {
-            foreach (var stmt in _preparedStatements)
+            if (DataReader != null)
             {
-                raw.sqlite3_reset(stmt);
-
-                yield return stmt;
+                DataReader.DisposeStatements = true;
+                DataReader = null;
```

Comment:
You are now depending on the Dispose method of the reader to be called to dispose the prepared statements. Before the base finalizer of the command did this everytime by calling Dispose(false).